File: BlocksWorld_SPD/PickUp.cs
using it.unical.mat.embasp.languages;
using System;
using System.Collections.Generic;
using System.Text;

namespace BlocksWorld_SPD
{
    [Id("edge")]
    class PickUp
    {
        [Param(0)]
        private string block;

        public PickUp()
        {
            this.block = "";
        }

        public PickUp(string block)
        {
            this.block = block;
        }

        public string getBlock()
        {
            return this.block;
        }

        public void setBlock(string block)
        {
            this.block = block;
        }

        public override string ToString()
        {
            return "pick-up(block = " + block + ")";
        }
    }
}

File: EmbASP_Desktop_BlocksWorld_Java/src/it/mat/unical/embasp/PickUp.java
package it.mat.unical.embasp;

import it.unical.mat.embasp.languages.Id;
import it.unical.mat.embasp.languages.Param;

@Id("pick-up")
public class PickUp {

	@Param(0)
	private String block;

	public PickUp() {
		block = new String();
	}

	public String getBlock() {
		return block;
	}

	public void setBlock(final String block) {
		this.block = block;
	}
	
	@Override
	public String toString() {
		return "pick-up(block = " + block + ")";
	}
}